block-storage/src/main/scala/coop/rchain/blockstorage/casperbuffer/CasperBufferStorage.scala

```diff
@@ -0,0 +1,24 @@
+package coop.rchain.blockstorage.casperbuffer
+
+import coop.rchain.blockstorage.util.{BlockDependencyDag, DoublyLinkedDag}
+import coop.rchain.models.BlockHash.BlockHash
+
+/**
+  * CasperBuffer manages block dependency graph.
+  * Block is in scope of Casper Buffer until it is validated and added to DAG.
+  */
+trait CasperBufferStorage[F[_]] {
+  def getParents(blockHash: BlockHash): F[Option[Set[BlockHash]]]
+  def getChildren(blockHash: BlockHash): F[Option[Set[BlockHash]]]
+  def getPendants: F[Set[BlockHash]]
+  def addRelation(parent: BlockHash, child: BlockHash): F[Unit]
+  def remove(blockHash: BlockHash): F[Unit]
+  def contains(blockHash: BlockHash): F[Boolean]
+  def isPendant(blockHash: BlockHash): F[Boolean]
+  def size: F[Long]
+  def toDoublyLinkedDag: F[DoublyLinkedDag[BlockHash]]
```

Comment:
`getParents`, `getChildren`, `size` and `isPendant` are not used. Do you have some plan for that?

They are also redundant because `toDoublyLinkedDag` gives all necessary info so these can be extension functions.

Or maybe from another perspective. Why to expose all internals with `DoublyLinkedDag[BlockHash]` and not just `getChildren` which is used here?
https://github.com/rchain/rchain/blob/36bf1e0f183ab04e925e0e49ca8c42537183c294/casper/src/main/scala/coop/rchain/casper/EquivocationDetector.scala#L52-L56

Reply:
@tgrospic yes I believe its better to have this methods on CasperBuffer then `blockBufferDependencyDag.parentToChildAdjacencyList.contains(block.blockHash) `
I'm not sure yet about usage of these methods but it looks natural to have them, so I've added

Reply:
Hard to follow whats the logic behind `requestedAsDependency`

Reply:
I'd like to leave it as it is for now. We will refactor later when working on Casper.